test/highlight/c/regular.c

```diff
@@ -2,6 +2,16 @@
 
 #define MACRO 0
 
+int c_init() { 1 }
+// the following aren't highlight correctly,
+// is this a problem with the parser?
+#define TESTMACRO (-1)
+#define lang_init() c_init()
+#define min(X,Y) ((X) < (Y) ? (X) : (Y))
```

Comment:
Yeah, I have noticed that `parenthesized_expression` is not captured at all if it belongs to a macro, not even by a simple query like `(parenthesized_expression) @container`.

Reply:
I think it might have something to do with the macro being parsed via an injected language, but I am not sure. I think I remember noticing something like this some time ago while looking at injected languages. I can try to look into this later and hopefully find a solution, but I think it might be a parser issue.

Reply:
The injected language is C itself. Try this minimal example:

```c
int main(int argc, char **argv) {
	return 0;
}
```

Now let's evaluate `vim.treesitter.get_parser():children()` and we get the empty table `{}`. Now let's add a preprocessor definition.

```c
#define PI 3.14

int main(int argc, char **argv) {
	return 0;
}
```

If we evaluate `vim.treesitter.get_parser():children()` again the language tree will have one child element of language `c`. C++ does something similar, except the injected language is `cpp`.

Reply:
Rust also uses a language injection of `rust` for macros, but highlighting works as expected for all cases I have seen there. Also, just to clarify, I did think it was injecting `c` code in the `#define` statement, I was just wandering if that is what is messing with the highlighting, but I haven't looked into this.